### PR TITLE
cleanup Fishbone-Moncrief disk

### LINKED ISSUE #1065: Fishbone-Moncrief Documentation should cite references
# Bug reports:

### Expected behavior:

It cites references.

### Current behavior:

It cites no references

### LINKED ISSUE #1270: Geometric quantities computed unsoundly
# Bug reports:

### Expected behavior:

Geometric quantities like `sin^2(theta)` are computed in a manner that prevents them from being negative.

### Current behavior:

For Fishbone-Moncrief (and possibly in other places in the code) the sin^2(theta) quantity is computed in a manner that means it can sometimes be negative. #1257 implements a simple hack that takes the absolute value after the computation, but it should be replaced with something more robust. In a comment on #1257 @wthrowe pointed out `sin^2(theta)` can be computed from:
sin(theta)^2 = X / r^2

where X is the nonnegative root of

0 = X^2 - (x^2 + y^2 - z^2 - a^2) X - (x^2 + y^2) z^2

I'm creating an issue for this because I don't know in how many other places this bug is present. @carmaza would you please try to take a look at this and correct it since you wrote the original code and thus know where else this is being done?

### PR DESCRIPTION
**Pull request by carmaza**
## Proposed changes

- Following discussions on other PRs, dimless spin $a$ is now the member variable to pass.
- Adds BibTeX bibliography to the documentation (should fix [#1065)](https://github.com/sxs-collaboration/spectre/issues/1065)
- Fixes a small bug when trying to obtain the Lorentz factor using the `variables` function
- Improves calculation of sin theta squared (should fix [#1270)](https://github.com/sxs-collaboration/spectre/issues/1270). A similar expression is also computed in `BondiHoyleAccretion`, but the quantity there is ensured to be positive.
- The `background_spacetime` getter and having `angular_momentum` be a member variable are needed for a `MagnetizedDisk` class I wrote that simply wraps most of the functions in `FishboneMoncriefDisk` before adding the magnetic field. 

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp [hunk 1]
```diff
@@ -44,10 +44,15 @@ namespace Solutions {
  * where \f$u^\mu\f$ is the 4-velocity. Then, all the fluid quantities are
  * assumed to share the same symmetries as those of the background spacetime,
  * namely they are stationary (independent of \f$t\f$), and axially symmetric
- * (independent of \f$\phi\f$). Self-gravity is neglected, so that the fluid
+ * (independent of \f$\phi\f$). Note that \f$r\f$ and \f$\theta\f$ can also be
+ * interpreted as Kerr (a.k.a. "spherical" Kerr-Schild) coordinates
+ * (see KerrSchildCoords), and that the symmetries of the equilibrium ensure
```
Comment (wthrowe) [inline src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp]: Based on the wording, I believe "KerrSchildCoords" was intended to be a link.  For some reason Doxygen is not generating the link automatically, but adding the `gr::` namespace prefix convinces it to do so.

File: docs/References.bib [hunk 2]
```diff
@@ -25,6 +25,18 @@ @article{Beckwith2011iy
   volume        = {193},
 }
 
+@article{Fishbone1976apj,
+  author  = {Fishbone, L. G. and Moncrief, V.},
+  title   = {Relativistic Fluid Disks in Orbit Around Kerr Black Holes},
```
Comment (wthrowe) [inline docs/References.bib]: `Kerr` -> `{Kerr}` (preserves capitalization)

File: docs/References.bib [hunk 3]
```diff
@@ -60,6 +72,17 @@ @article{Komissarov1999
   pages   = {343--366},
 }
 
+@article{Kozlowski1978aa,
+  author  = {Kozlowski, M. and Jaroszynski, M. and Abramowicz, M. A.},
+  title   = {The analytic theory of fluid disks orbiting the Kerr black hole},
```
Comment (wthrowe) [inline docs/References.bib]: `Kerr` -> `{Kerr}`

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp [hunk 4]
```diff
@@ -119,35 +124,18 @@ namespace Solutions {
...
- * x &= \sqrt{r^2 + a^2}\sin\theta\cos\phi\\
- * y &= \sqrt{r^2 + a^2}\sin\theta\sin\phi\\
- * z &= r\cos\theta
- * \f}
- *
- * helps read the Jacobian matrix, which, applied to the azimuthal flow of the
- * disk, gives
+ * Once all the variables are known in Boyer-Lindquist (or Kerr) coordinates, it
+ * is straightforward to write them in Cartesian Kerr-Schild coordinates. The
+ * coordinate transformation in KerrSchildCoords helps read the Jacobian matrix,
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp]: is this also supposed to be a reference?

**Review by kidder — COMMENTED** [review]
Looks good after the minor fixes are made

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
@wthrowe please look at

**Review by wthrowe — DISMISSED** [review]

**Comment by kidder** [comment]
@carmaza could you please add a test that would have failed with negative sin^2

**Comment by carmaza** [comment]
I included the test in the last commit. The rest of this PR remains the same.

**Review by kidder — DISMISSED** [review]

**Comment by wthrowe** [comment]
Some versions of Catch fail tests that contain no assertions.  Add a `CHECK(true)` or something.

**Comment by carmaza** [comment]
Done!

**Comment by kidder** [comment]
@wthrowe this has been updated

**Review by kidder — DISMISSED** [review]

**Review by wthrowe — DISMISSED** [review]

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp [hunk 5]
```diff
@@ -166,27 +166,26 @@ class FishboneMoncriefDisk {
...
-  struct BlackHoleMass {
+  /// The mass of the black hole, \f$M\f$.
+  struct BhMass {
     using type = double;
     static constexpr OptionString help = {"The mass of the black hole."};
     static type lower_bound() noexcept { return 0.0; }
   };
-  /// The dimensionless black hole spin magnitude.
-  struct BlackHoleSpin {
+  /// The dimensionless black hole spin, \f$a_* = a/M\f$.
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp]: The dimensionless spin is usually denoted with Chi in SXS literature

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp [hunk 6]
```diff
@@ -317,6 +317,10 @@ class FishboneMoncriefDisk {
     return equation_of_state_;
   }
 
+  const gr::Solutions::KerrSchild& background_spacetime() const noexcept {
```
Comment (nilsdeppe) [inline src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp]: Why is this added? It's not needed

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/Test_FishboneMoncriefDisk.cpp [hunk 7]
```diff
@@ -162,6 +188,13 @@ SPECTRE_TEST_CASE("Unit.PointwiseFunctions.AnalyticSolutions.RelEuler.FMDisk",
   test_variables(DataVector(5));
 }
 
+SPECTRE_TEST_CASE(
```
Comment (nilsdeppe) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/Test_FishboneMoncriefDisk.cpp]: Please write this into a function in an anonymous namespace and call that from an existing test case. In general, we need to reduce the number of `SPECTRE_TEST_CASE`s by doing this sort of refactoring. There is a non-negligible overhead to running a test case that we need to try and reduce more.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
I'm happy for you to squash the changes

File: src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp (hunk 6 above)
Comment (carmaza) [reply src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp]: it's not need now, but it'll be used for retrieving the `background_spacetime` for the magnetized disk. Since the later is just a FM disk + a B-field, I find retrieving the background spacetime from `FishboneMoncriefDisk` more consistent than hard-coding the same background spacetime in the magnetized FM class. If for some reason this sounds unnecessary or stupid, I can leave this commit out.

Comment (nilsdeppe) [reply src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp]: I don't see why it's ever needed. You can always grab the spacetime solution from the `variables` function. We already needed this for the evolution. This function was explicitly removed from all the analytic solutions so that all variables are treated on equal footing.

Comment (carmaza) [reply src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp]: commit was removed

**Comment by carmaza** [comment]
squashed and rebased on develop

**Review by nilsdeppe — DISMISSED** [review]

**Review by wthrowe — DISMISSED** [review]

**Comment by kidder** [comment]
@carmaza please rebase as we updated Travis

**Comment by carmaza** [comment]
rebased after Travis update

**Review by wthrowe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Comment by nilsdeppe** [comment]
LGTM, travis failures GCC6/7 debug builds, the usual segfaults.
